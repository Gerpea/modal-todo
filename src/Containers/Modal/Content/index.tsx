import React from 'react';

import style from '../styles.module.css';

export interface Props {
  children: React.ReactNode;
}

function ModalContent({ children }: Props): React.ReactElement {
  return <div className={style.Content}>{children}</div>;
}

export default ModalContent;
