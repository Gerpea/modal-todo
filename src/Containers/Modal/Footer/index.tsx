import React from 'react';

import style from '../styles.module.css';

export interface Props {
  children: React.ReactNode;
}

function ModalFooter({ children }: Props): React.ReactElement {
  return <div className={style.Footer}>{children}</div>;
}

export default ModalFooter;
