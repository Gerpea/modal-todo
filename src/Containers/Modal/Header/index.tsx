import React from 'react';

import style from '../styles.module.css';

export interface Props {
  children: React.ReactNode;
}

function ModalHeader({ children }: Props): React.ReactElement {
  return <div className={style.Header}>{children}</div>;
}

export default ModalHeader;
