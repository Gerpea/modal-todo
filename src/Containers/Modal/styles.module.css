.Modal {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);

  background-color: var(--c-gray-1);
  box-shadow: 0px 20px 50px rgba(0, 0, 0, 0.25);

  padding: 20px;

  overflow: hidden;
  -webkit-overflow-scrolling: touch;
  outline: none;
}

.Header {
  padding-bottom: 10px;
  border-bottom: 2px solid var(--c-gray-2);
  font-weight: bold;
}

.Content {
  padding: 20px 0;
}

.Footer {
  padding-top: 19px;
  border-top: 2px solid var(--c-gray-2);
}

@media (max-width: 600px) {
  .Modal {
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;

    width: auto;
    height: auto;

    transform: none;
  }
}
