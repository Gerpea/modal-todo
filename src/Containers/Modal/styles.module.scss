@import '../../variables';

.Modal {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  row-gap: 21px;

  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);

  background-color: $c-gray-1;
  box-shadow: 0px 20px 50px rgba($c-black, 0.25);

  padding: 20px;

  overflow: hidden;
  -webkit-overflow-scrolling: touch;
  outline: none;
}

@media (max-width: 600px) {
  .Modal {
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;

    width: auto !important;
    height: auto !important;

    transform: none;
  }
}
