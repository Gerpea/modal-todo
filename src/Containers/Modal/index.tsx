import React from 'react';
import ReactModal from 'react-modal';

import Header from './Header';
import Footer from './Footer';
import Content from './Content';

import style from './styles.module.css';
export interface Props {
  isOpen: boolean;
  title: string;
  onClose: () => void;
  footer?: React.ReactNode;
  children: React.ReactNode;
}

function Modal({ isOpen, title, onClose, footer, children }: Props): React.ReactElement {
  return (
    <ReactModal isOpen={isOpen} className={style.Modal} onRequestClose={onClose}>
      <Header>{title}</Header>
      <Content>{children}</Content>
      {footer && <Footer>{footer}</Footer>}
    </ReactModal>
  );
}

export default Modal;
