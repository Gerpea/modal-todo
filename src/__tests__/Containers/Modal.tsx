import React, { ReactPortal } from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Modal from 'Containers/Modal';

describe('Modal test', () => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  let originalCreatePortal: any;
  beforeAll(() => {
    originalCreatePortal = ReactDOM.createPortal;
    ReactDOM.createPortal = (node) => node as ReactPortal;
  });

  afterAll(() => {
    ReactDOM.createPortal = originalCreatePortal;
  });

  it('Modal open', () => {
    const tree = renderer.create(<Modal isOpen={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Modal close', () => {
    const tree = renderer.create(<Modal isOpen={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
