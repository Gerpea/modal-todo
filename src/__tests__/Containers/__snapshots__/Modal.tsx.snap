// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal test Modal close 1`] = `null`;

exports[`Modal test Modal open 1`] = `
<div
  className="ReactModal__Overlay"
  onClick={[Function]}
  onMouseDown={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 0.75)",
      "bottom": 0,
      "left": 0,
      "position": "fixed",
      "right": 0,
      "top": 0,
    }
  }
>
  <div
    aria-modal={true}
    className="ReactModal__Content"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    role="dialog"
    style={
      Object {
        "WebkitOverflowScrolling": "touch",
        "background": "#fff",
        "border": "1px solid #ccc",
        "borderRadius": "4px",
        "bottom": "40px",
        "left": "40px",
        "outline": "none",
        "overflow": "auto",
        "padding": "20px",
        "position": "absolute",
        "right": "40px",
        "top": "40px",
      }
    }
    tabIndex="-1"
  />
</div>
`;
